File: lat2/0.py
import pandas as pd
import mysql.connector
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
df = pd.read_sql(query1, connection)
connection.close()

df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df = df[df['order_date'].dt.year == 2021]
df['month'] = df['order_date'].dt.month 
print(df)

tabel1 = df.groupby('month')['qty_ordered'].sum().reset_index(name='kuantitas')
tabel2 = df.groupby('month')[['customer_id', 'id']].nunique().reset_index()
tabel = pd.merge(tabel1, tabel2, on='month', how='inner')
tabel = tabel.sort_values('kuantitas', ascending=False).reset_index(drop=True)
print(tabel)

File: lat2/0.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query <- "SELECT * FROM order_detail"
df <- dbGetQuery(con, query)
dbDisconnect(con)

print(head(df))

df <- filter(df, is_valid == 1)
df$order_date <- as.Date(df$order_date)
df <- df[year(df$order_date) == 2021, ]
df$month <- month(df$order_date)
print(head(df))


tabel1 <- df %>% group_by(month) %>% summarise(kuantitas = sum(qty_ordered))
tabel2 <- df %>% group_by(month) %>% summarise(customer_id = n_distinct(customer_id), id = n_distinct(id))
tabel <- merge(tabel1, tabel2, by = 'month')
tabel <- tabel %>% arrange(desc(kuantitas))
print(tabel)